snuba_sdk/expressions.py

```diff
@@ -23,9 +23,45 @@ def validate(self) -> None:
 
 
 # For type hinting
-ScalarType = Union[None, bool, str, bytes, float, int, date, datetime]
+ScalarLiteralType = Union[None, bool, str, bytes, float, int, date, datetime]
+ScalarSequenceType = Sequence[ScalarLiteralType]
+ScalarType = Union[ScalarLiteralType, ScalarSequenceType]
+
 # For type checking
-Scalar: Set[type] = {type(None), bool, str, bytes, float, int, date, datetime}
+Scalar: Set[type] = {
+    type(None),
+    bool,
+    str,
+    bytes,
+    float,
+    int,
+    date,
+    datetime,
+}
+
+
+class InvalidArray(Exception):
+    def __init__(self, value: List[Any]) -> None:
+        value_str = f"{value}"
+        if len(value_str) > 10:
+            value_str = f"{value_str[:10]}...]"
+
+        super().__init__(
+            f"invalid array {value_str}: arrays must have the same data type or None, perhaps use a tuple instead"
+        )
+
+
+def is_scalar(value: Any) -> bool:
+    if isinstance(value, tuple(Scalar)) or isinstance(value, tuple):
```

Comment:
Why does `isinstance(value, tuple)`  make the method return true? a Tuple is not a scalar.

Reply:
It is now, that's what this PR adds. list and tuple are now valid scalar types, as long as they themselves contain scalars. I will add a check for that, since I have it for the array.